hw/ip/spi_host/doc/_index.md

```diff
@@ -0,0 +1,633 @@
+
+Devices also differ in their expectations of clock *phase* (CPHA) relative to the data.
+Devices with CPHA = 0, expect that both the host and device will be sampling data on the *leading* edge, and asserting data on the *trailing* edge.
+(Because of the option for either polarity, the terms "leading" and "trailing" are preferred to "rising" or "falling").
+When CPHA = 0, the first output bit is asserted with the falling edge of `cs_n`.
+Meanwhile if CPHA = 1, data is always is asserted on the leading edge of `sck`, and data is always sampled on the trailing edge of sck.
+
+When operating at the fastest-rated clock speeds, some flash devices (i.e. both the Winbond and Micron devices noted above), require setup times which exceed half a clock-cycle.
+In order to support these fastest data rates, the SPI_HOST IP offers a modified "Full-cycle" (FULLCYC = 1) timing mode where data can be sampled a *full* cycle after the target device asserts data on the `sd` bus.
+This full cycle mode has no effect on any of the signals transmitted, only on the timing of the sampling of the incoming signals.
+
+{{< wavejson >}}
+{signal: [
+  {name: "clk_core_i", wave: "p.................."},
+  {name: "sck (CPOL=0)", wave: "0.1010101010101010."},
+  {name: "sck (CPOL=1)", wave: "1.0101010101010101."},
+  {name: "cs_n", wave: "10................1"},
+  ["CPHA = 0",
+    {name: "sd[0] (output)", wave: "22.2.2.2.2.2.2.2...", data: ["","out7", "out6", "out5", "out4", "out3", "out2", "out1", "out0" ]},
+    {name: "sd[1] (input)", wave: "22.2.2.2.2.2.2.2.2.", data: ["","in7", "in6", "in5", "in4", "in3", "in2", "in1", "in0",""]},
+    {name: "Sampling Event (FULLCYC=0)", wave: "1.H.H.H.H.H.H.H.H.."},
+    {name: "Sampling Event (FULLCYC=1)", wave: "1..H.H.H.H.H.H.H.H."},
+  ],
+  ["CPHA = 1",
+    {name: "sd[0] (output)", wave: "2.2.2.2.2.2.2.2.2..", data: ["","out7", "out6", "out5", "out4", "out3", "out2", "out1", "out0" ]},
+    {name: "sd[1] (input)", wave: "2.2.2.2.2.2.2.2.2.2", data: ["","in7", "in6", "in5", "in4", "in3", "in2", "in1", "in0",""]},
+    {name: "Sampling Event (FULLCYC=0)", wave: "1..H.H.H.H.H.H.H.H."},
+    {name: "Sampling Event (FULLCYC=1)", wave: "1...H.H.H.H.H.H.H.H"},
+  ],
+],
+  head: {
+    text: "Standard SPI transaction (1 byte) illustrating of the impact of the CPOL, CPHA, and FULLCYC settings"
+  },
+  foot: {
+  }
+}
+{{< /wavejson >}}
+
+
+In the faster Dual- or Quad-modes, all data lines are bidirectional, and in Quad mode the number of data lines increases to four.
+For the purposes of this IP, Dual or Quad-mode commands can be thought of as consisting of up to four command phases in which the host:
+1. Transmits instructions or data at single-line rate,
+2. Transmits instructions address or data on 2 or 4 data lines,
+3. Holds the bus in a high resistance state for some number of "dummy" clock cycles (neither side transmits), or
```

Comment:
would it be better to say high impedance?

Reply:
Yes.  Agreed.  Thanks for catching that.  Changed.